# reasonable parameters to find planes of @Work workspaces
multimodal_object_recognition:
  voxel_leaf_size: 0.009
  voxel_filter_field_name: z
  voxel_filter_limit_min: -0.15
  voxel_filter_limit_max: 0.3
  enable_passthrough_filter: false
  passthrough_filter_field_name: x
  passthrough_filter_limit_min: 0.0
  passthrough_filter_limit_max: 0.8
  enable_cropbox_filter: true
  cropbox_filter_min_x: 0.2
  cropbox_filter_max_x: 0.8
  cropbox_filter_min_y: -0.4
  cropbox_filter_max_y: 0.4
  cropbox_filter_min_z: -0.1
  cropbox_filter_max_z: 0.6
  normal_radius_search: 0.03
  use_omp: False
  num_cores: 8
  sac_max_iterations: 1000
  sac_distance_threshold: 0.01
  sac_optimize_coefficients: true
  sac_x_axis: 0.0
  sac_y_axis: 0.0
  sac_z_axis: 1.0
  sac_eps_angle: 0.09
  sac_normal_distance_weight: 0.05
  prism_min_height: 0.01
  prism_max_height: 0.1
  outlier_radius_search: 0.03
  outlier_min_neighbors: 20
  cluster_tolerance: 0.02
  cluster_min_size: 25
  cluster_max_size: 20000
  cluster_min_height: 0.011
  cluster_max_height: 0.09
  cluster_max_length: 0.25
  cluster_min_distance_to_polygon: 0.04
  center_cluster: False
  pad_cluster: True
  padded_cluster_size: 2048
  octree_resolution: 0.0025
  height_of_floor: -0.083
  use_fixed_heights: False
  object_height_above_workspace: -0.008
  container_height: 0.07
  enable_rgb_recognizer: true
  enable_pc_recognizer: false
  obj_category: 'atwork' # atwork, cavity, container
  rgb_roi_adjustment: 2
  rgb_bbox_min_diag: 21
  rgb_bbox_max_diag: 200
  rgb_cluster_filter_limit_min: 0.005 # for tower camera configuration (youbot 2)
  rgb_cluster_filter_limit_max: 0.35
  rgb_cluster_remove_outliers: True
  enable_roi: True
  roi_base_link_to_laser_distance: 0.350
  roi_max_object_pose_x_to_base_link: 0.700
  roi_min_bbox_z: 0.03
